x-pack/plugins/security_solution/server/lib/timeline/utils/migrator.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from '@elastic/safer-lodash-set';
+import _ from 'lodash';
+import { SavedObject, SavedObjectReference, SavedObjectsUpdateResponse } from 'kibana/server';
+
+interface Field {
+  path: string;
+  type: string;
+  name: string;
+}
+
+export class FieldMigrator {
```

Comment:
Can we put a note in this file that essentially says the resetting of the object to its original state in the `populateFields*` calls is to avoid making wholesale changes on the UI side as to the expectations of the shape of the data?